flytekit/models/task.py

```diff
@@ -383,6 +394,10 @@ def to_flyte_idl(self):
             container=self.container.to_flyte_idl() if self.container else None,
             task_type_version=self.task_type_version,
         )
+        if self.config is not None:
+            for k, v in self.config.items():
+                task_template.config[k] = v
```

Comment:
is task_template not of type dict? are we just trying to make a copy of it?, can we not do this:
```suggestion
        config={k: v for k, v in self.config.items()} if self.config is not None else None,
    )
```

Reply:
oh this doesn't work because the config is a proto map so I can only write key by key